lint/util.py

```diff
@@ -44,18 +44,20 @@ def printf(*args):
 
 
 def get_syntax(view):
-    """
-    Return the view's syntax.
-
-    or the syntax it is mapped to in the "syntax_map" setting.
-    """
+    """Return the view's syntax."""
     from .persist import settings
     syntax_map = settings.get('syntax_map')
-    for syntax, selectors in syntax_map.items():
-        for s in selectors:
-            if view.match_selector(0, s):
-                return syntax
-    return ""
+    try:
+        scoring = {
+            max(view.score_selector(0, s) for s in selectors if s): syntax
```

Comment:
If the iterable for `max` is of zero length, it raises a `ValueError`, but I think we should handle this case by providing a `default` kwarg.

Reply:
Which one?

Reply:
well, if `selectors` is empty, `max` will err out. But this is the inner max for the entries' selectors, not as a whole, so when there's an empty list for one map, the function will always return `""`.

The solution would be to invoke `max(iterable, default=0)`. Regardless, this will only happen for invalid configurations (i.e. not having a single selector entry in the array).